src/OpenTelemetry/Metrics/Exemplar/AlignedHistogramBucketExemplarReservoir.cs

```diff
@@ -8,92 +8,28 @@ namespace OpenTelemetry.Metrics;
 /// <summary>
 /// The AlignedHistogramBucketExemplarReservoir implementation.
 /// </summary>
-internal sealed class AlignedHistogramBucketExemplarReservoir : ExemplarReservoir
+internal sealed class AlignedHistogramBucketExemplarReservoir : FixedSizeExemplarReservoir
 {
-    private readonly Exemplar[] runningExemplars;
-    private readonly Exemplar[] tempExemplars;
-
-    public AlignedHistogramBucketExemplarReservoir(int length)
-    {
-        this.runningExemplars = new Exemplar[length + 1];
-        this.tempExemplars = new Exemplar[length + 1];
-    }
-
-    public override void Offer(long value, ReadOnlySpan<KeyValuePair<string, object?>> tags, int index = default)
-    {
-        this.OfferAtBoundary(value, tags, index);
-    }
-
-    public override void Offer(double value, ReadOnlySpan<KeyValuePair<string, object?>> tags, int index = default)
+    public AlignedHistogramBucketExemplarReservoir(int numberOfBuckets)
+        : base(numberOfBuckets + 1)
     {
-        this.OfferAtBoundary(value, tags, index);
     }
 
-    public override Exemplar[] Collect(ReadOnlyTagCollection actualTags, bool reset)
+    public override void Offer(in ExemplarMeasurement<long> measurement)
     {
-        for (int i = 0; i < this.runningExemplars.Length; i++)
-        {
-            this.tempExemplars[i] = this.runningExemplars[i];
-            if (this.runningExemplars[i].FilteredTags != null)
-            {
-                // TODO: Better data structure to avoid this Linq.
-                // This is doing filtered = alltags - storedtags.
-                // TODO: At this stage, this logic is done inside Reservoir.
-                // Kinda hard for end users who write own reservoirs.
-                // Evaluate if this logic can be moved elsewhere.
-                // TODO: The cost is paid irrespective of whether the
-                // Exporter supports Exemplar or not. One idea is to
-                // defer this until first exporter attempts read.
-                this.tempExemplars[i].FilteredTags = this.runningExemplars[i].FilteredTags!.Except(actualTags.KeyAndValues.ToList()).ToList();
-            }
-
-            if (reset)
-            {
-                this.runningExemplars[i].Timestamp = default;
-            }
-        }
+        Debug.Assert(
+            measurement.ExplicitBucketHistogramBucketIndex != -1,
+            "ExplicitBucketHistogramBucketIndex was -1");
 
-        return this.tempExemplars;
+        this.UpdateExemplar(measurement.ExplicitBucketHistogramBucketIndex, in measurement);
```

Comment:
We don't have to override this method. For histograms, `ExemplarReservoir.Offer` is only called for `double` values.

Reply:
We should probably leave the implementation blank then just so that it's clear we don't expect this method to be called.